avocado/utils/process.py

```diff
@@ -212,33 +212,37 @@ def kill_process_tree(pid, sig=signal.SIGKILL, send_sigcont=True,
     :param timeout: How long to wait for the pid(s) to die
                     (negative=infinity, 0=don't wait,
                     positive=number_of_seconds)
+    :return: killed_pids that were destroyed
     """
-    if timeout <= 0:
-        def remaining_time():
-            return timeout
-    else:
-        start = time.time()
+    def _all_pids_dead(killed_pids):
+        for pid in killed_pids:
+            if pid_exists(pid):
+                return False
+        return True
 
-        def remaining_time():
-            return timeout - time.time() + start
+    killed_pids = [pid]
+    if timeout > 0:
+        start = time.time()
 
     if not safe_kill(pid, signal.SIGSTOP):
-        return
+        return killed_pids
     for child in get_children_pids(pid):
-        kill_process_tree(int(child), sig, timeout=remaining_time())
+        killed_pids.extend(kill_process_tree(int(child), sig, False))
     safe_kill(pid, sig)
     if send_sigcont:
-        safe_kill(pid, signal.SIGCONT)
+        for pid in killed_pids:
+            safe_kill(pid, signal.SIGCONT)
     if timeout == 0:
-        return
+        return killed_pids
```

Comment:
Shouldn't we be testing the now different return value?  I mean, I don't see the unittests reflecting/checking that change.

Reply:
sure